util/GameRules.h

```diff
@@ -100,9 +102,17 @@ class FO_COMMON_API GameRules {
         auto it = m_game_rules.find(name);
         if (it != m_game_rules.end())
             throw std::runtime_error("GameRules::Add<>() : Rule " + name + " was added twice.");
-        m_game_rules[name] = Rule(RuleTypeForType(T()), name, default_value, default_value,
-                                  description, validator.Clone(), engine_interal, category);
-        DebugLogger() << "Added game rule named " << name << " with default value " << default_value;
+        if (!GetOptionsDB().OptionExists("setup.rules.server-locked." + name)) {
+            GetOptionsDB().Add<bool>("setup.rules.server-locked." + name, description, false);
```

Comment:
Is there any security risk if the `server-locked` rules exist on the client and server, if the client sends lobby data that contains a `server-locked` rule set to false or true? Could a modified client add aa rule with `server-locked` in the name, even if one doesn't exist on the server, and thereby cause the lobby data sent to the server to copy that rule into the server's rules, and thus lock another rule that wasn't supposed to be locked on the server?

Reply:
`server-locked.*` is an option, not an game rule. It isn't transferred between client and server as I know.

Reply:
Debug level options do transfer, but that is set up explicitly in the logging options UI code and has a dedicated client server message type.